### DISCUSSION TITLE
Lunar Nodes

### DISCUSSION THREAD

**Discussion by jtdaly4** [cosinekitty/astronomy#223] [Q&A] [answered]
Are the Lunar Nodes available on your Api? If not, could we use your data to calculate the nodes? 
thank you

**Comment by cosinekitty** [comment, marked as answer]
Yes, there are two functions for searching for consecutive lunar nodes. I don't know which programming language you are interested in, but here they are in the C# version. The other languages have similar function names.

- [SearchMoonNode](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#Astronomy.SearchMoonNode)
- [NextMoonNode](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#Astronomy.NextMoonNode)

Does this help?

**Reply by jtdaly4** [reply]
of course! thats fantastic news. we will be utilizing your work asap! great job. thank you

**Comment by cosinekitty** [comment]
To find lunar eclipses, try using [SearchLunarEclipse](https://github.com/cosinekitty/astronomy/tree/master/source/python#SearchLunarEclipse) and [NextLunarEclipse](https://github.com/cosinekitty/astronomy/tree/master/source/python#NextLunarEclipse).

SearchLunarApsis finds perigee and apogee, the times the Moon is closest to the Earth and farthest from the Earth, which can happen at any lunar phase, so they can't be used to determine eclipses.

Lunar nodes are a third concept, where the Moon is ascending or descending through the ecliptic plane. You can call [SearchMoonNode](https://github.com/cosinekitty/astronomy/tree/master/source/python#SearchMoonNode) to find the time and type of node. Then call [EclipticLongitude](https://github.com/cosinekitty/astronomy/tree/master/source/python#EclipticLongitude) using that time.

I hope this helps!

**Reply by xeno-sama** [reply]
thx
got it
